python/src/foxglove_websocket/server/__init__.py

```diff
@@ -303,29 +487,23 @@ async def _handle_client_message(self, client: ClientState, message: ClientMessa
                         chan_id,
                     )
                     if self._listener and first_subscription:
-                        self._listener.on_subscribe(self, chan_id)
+                        await self._listener.on_subscribe(self, chan_id)
```

Comment:
Didn't get this working, will give it another try tomorrow

Reply:
Maybe we should write up some notes on how to migrate to the new version.
Actually it would be nice if we can detect the old implementations (using iscoroutinefunction) and throw an error if we can't handle it properly?